plugins/device/cmd/nvidia/nvml_client/client.go

```diff
@@ -0,0 +1,194 @@
+package nvml
+
+import (
+	"fmt"
+)
+
+// DeviceData represents common fields for Nvidia device
+type DeviceData struct {
+	UUID       string
+	DeviceName *string
+	MemoryMiB  *uint64
+	PowerW     *uint
+	BAR1MiB    *uint64
+}
+
+// FingerprintDeviceData is a superset of DeviceData
+// it describes device specific fields returned from
+// nvml queries during fingerprinting call
+type FingerprintDeviceData struct {
+	*DeviceData
+	PCIBandwidthMBPerS *uint
+	CoresClockMHz      *uint
+	MemoryClockMHz     *uint
+	DisplayState       string
+	PersistenceMode    string
+	PCIBusID           string
+}
+
+// FingerprintData represets attributes of driver/devices
+type FingerprintData struct {
+	Devices       []*FingerprintDeviceData
+	DriverVersion string
+}
+
+// StatsData is a superset of DeviceData
+// it represents statistics data returned for every Nvidia device
+type StatsData struct {
+	*DeviceData
+	PowerUsageW        *uint
+	GPUUtilization     *uint
+	MemoryUtilization  *uint
+	EncoderUtilization *uint
+	DecoderUtilization *uint
+	TemperatureC       *uint
+	UsedMemoryMiB      *uint64
+	BAR1UsedMiB        *uint64
+	ECCErrorsL1Cache   *uint64
+	ECCErrorsL2Cache   *uint64
+	ECCErrorsDevice    *uint64
+}
+
+// NVMLClientInterface describes how users would use nvml library
+type NVMLClientInterface interface {
```

Comment:
Generally don't put the word interface in the interface. Just call it `NVMLClient`